doc/mod_wsgi.md

```diff
@@ -0,0 +1,200 @@
+# Running SATOSA Using Apache HTTP Server and mod\_wsgi
+
+This document describes how to deploy and run the SATOSA proxy using
+Apache HTTP Server and mod\_wsgi. To be concrete this document details
+deploying SATOSA on the latest CentOS 7.x release.
+
+## Dependencies
+
+```
+yum install epel-release
+yum install httpd mod_ssl httpd-devel python34 python34-devel
+yum install xmlsec1-openssl gcc curl
+```
+
+Install the latest production release of pip and use it to install the latest 
+production release of mod\_wsgi:
+
+```
+curl https://bootstrap.pypa.io/get-pip.py | python3
+pip install mod_wsgi
+```
+
+Create a `satosa` user to run the WSGI daemon:
+
+```
+useradd --home-dir /etc/satosa --no-create-home --system --user-group satosa
+```
+
+## Installation
+
+Use pip to install SATOSA:
+
+```
+pip install SATOSA
+```
+
+To instead install the latest from the master branch on the GitHub repository:
+
+```
+yum install git
+pip install --upgrade git+https://github.com/IdentityPython/SATOSA.git#egg=SATOSA
+```
+
+To upgrade and use the latest release of pySAML2:
+
+```
+pip install --upgrade pysaml2
+```
+
+## Installation of SATOSA Microservices
+
+```
+curl -L -o satosa_microservices.tar.gz \
+    https://github.com/IdentityPython/satosa_microservices/archive/master.tar.gz \
+    && mkdir -p /opt/satosa_microservices \
+    && tar -zxf satosa_microservices.tar.gz -C /opt/satosa_microservices --strip-components=1 \
+    && rm -f satosa_microservices.tar.gz
+```
+
+If you need the LDAP Attribute Store microservice you must also install
+ldap3 using pip:
+
+```
+pip install ldap3
+```
+
+## Apache Configuration
+
+Use the `mod_wsgi-express module-config` command to determine the correct
+module path and Python home to add to the Apache configuration. For
+example:
+
+```
+$ mod_wsgi-express module-config
+LoadModule wsgi_module "/usr/lib64/python3.4/site-packages/mod_wsgi/server/mod_wsgi-py34.cpython-34m.so"
+WSGIPythonHome "/usr"
+```
+
+Edit the Apache config and in the global section (not within a virtual
```

Comment:
This seems to be needed because `mod_wsgi` is not managed by the distribution. I think that if you install mod_wsgi through `yum` this can be skipped. Apache provides tools to automatically (un)load modules with `a2enmod` (and `a2dismod`) which should be automatically invoked by the mod_wsgi installation by the package manager.

Reply:
Again the version packaged and available through yum has significant bugs.